### ISSUE TITLE
the sample code "translate.py" does not train a correct model

### ISSUE THREAD

**Issue by xuancong84** [tensorflow/tensorflow#5922]
After v0.10, the seq2seq sample code, translate.py, although runnable, but does not give correct output.
I trained a default model using downloaded EN->FR data set, the trained system cannot even translate basic words like: I, you, he and she.
Does someone know why?

Below are my system info:
    Operating System:
    Ubuntu 14.04.5 LTS

    Installed version of CUDA and cuDNN:
    /usr/local/cuda-8.0
    cudnn-8.0-linux-x64-v5.1.tgz

    It is installed from binary pip package

    A link to the pip package you installed:
    https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.11.0-cp27-none-linux_x86_64.whl

    The output from python -c "import tensorflow; print(tensorflow.version)"
    xuancong@wxc-i2r:~/projects/tf-rnnlm$ python -c "import tensorflow; print(tensorflow.version)"
    I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
    I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so locally
    I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
    I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
    I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
    0.11.0

**Comment by poxvoculi** [comment]
This sort of question is better asked on stackoverflow, assuming that it's user error and not a completely broken model.  In order to try to make that distinction, it would be helpful if you would clarify what happened when you trained the model, so we can determine whether training failed, or inference from a correct model is not working.

Did the training run the expected number of steps without error?  Did the loss decrease as expected?  Did you run evaluations on the training set and a hold-out set and see the expected increases in accuracy during training?    If the training failed, you need to figure out why.

If training succeed, then the problem must lie in inference.  Again, it would be helpful to know exactly how you invoked inference and what diagnostic messages might have occurred.

**Comment by xuancong84** [comment]
@poxvoculi Thanks for your reply!
Since I am new to NN, there is something I would like to clarify with you.
Since our server does not have enough graphics memory, I reduced the embedding size from 1024 to 512 to fit into memory.
Since training took forever, the one I tested is at about:
global step 150200 learning rate 0.1262 step-time 0.51 perplexity 6.82
  eval: bucket 0 perplexity 3.91                                      
  eval: bucket 1 perplexity 5.67                                      
  eval: bucket 2 perplexity 6.33                                      
  eval: bucket 3 perplexity 11.45                                     

All other parameters I use the default.
So does everything look normal?

**Comment by poxvoculi** [comment]
Those are some critical details.  By reducing the embedding size, you reduced its ability to distinguish between words dramatically.  Also, you may not have trained long enough.  The tutorial suggests you should train for a full epoch (340,000 steps at batch 64) before expecting the model to work...with the original embedding size.  Nonetheless, the perplexity scores suggest that it learned something. 

One rationality test would be to retry queries from your eval set on the inference version you say cannot translate basic terms.  Do you get the same overall perplexity score as at the end of training?  If not, if its much worse, then you've somehow set up the inference model incorrectly.  If so, then either your sample test queries are not good (try full sentences, not single words), or the model isn't capable enough or trained long enough.

I'm not an expert on translation models.  Perhaps someone else reading this thread has more insight.

**Comment by michaelisard** [comment]
Closing due to lack of activity.
